### ISSUE TITLE
Add Sidebar "always on" experiment metrics

### ISSUE THREAD

**Issue by DJAndries** [brave/brave-browser#36815]
Adding metrics for the Sidebar "always on" experiment (https://github.com/brave/brave-browser/issues/36339):

As a "sidebar enabled" experiment participant, do you have Leo enabled?
`Brave.AIChat.Enabled.SidebarEnabledA` or `Brave.AIChat.Enabled.SidebarEnabledB`

As a "sidebar enabled" experiment participant, did you use Leo in the past day?
`Brave.AIChat.UsageDaily.SidebarEnabledA` or `Brave.AIChat.UsageDaily.SidebarEnabledB`

As a "sidebar enabled" experiment participant, did you use Leo in the past week?
`Brave.AIChat.UsageWeekly.SidebarEnabledA` or `Brave.AIChat.UsageWeekly.SidebarEnabledB`

The answers for these metrics would be:

1. Yes, as a free user
2. Yes, as a premium user

As a "sidebar enabled" experiment participant, did you change the Sidebar settings?
`Brave.Core.SidebarSettings.SidebarEnabledA` or `Brave.Core.SidebarSettings.SidebarEnabledB`

0. No
1. Yes, I changed Show Sidebar to "On mouse over"
2. Yes, I changed Show Sidebar to "Never"

cc @mattmcalister

**Comment by kjozwiak** [comment]
Removing from the milestone as https://github.com/brave/brave-core/pull/22615 hasn't been merged into `master`. Once https://github.com/brave/brave-core/pull/22615 lands into `master`, it will be added into the appropriate milestone and then can be moved once uplifted.

**Comment by kjozwiak** [comment]
The above requires `1.65.111` or higher for `1.65.x` verification 👍

**Comment by GeetaSarvadnya** [comment]
Verification INPROGRESS on

```
Brave | 1.65.112 Chromium: 124.0.6367.29 (Official Build) (64-bit)
-- | --
Revision | 1ccdfe627ad458201908007edf699da1f3bfc02b
OS | Windows 10 Version 22H2 (Build 19045.4170)
```

**Comment by GeetaSarvadnya** [comment]
Verification **INPROGRESS** on `Samsung Galaxy version 12` running `Bravemonoarm64.apk_1.65.114`
